src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,19 +6,74 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  ContentChildren,
+  Directive,
+  EventEmitter, forwardRef, Inject,
+  Input, Output,
+  QueryList
+} from '@angular/core';
+import {BooleanInput, coerceBooleanProperty} from '@angular/cdk/coercion';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '(click)': 'toggle()',
+    '[attr.aria-selected]': 'selected || null',
+    '[attr.id]': 'id',
   }
 })
 export class CdkOption {
+  private _selected: boolean = false;
+  readonly _uniqueid: string;
+  private _id: string;
+
+  /** Whether the option is selected or not */
+  @Input()
+  get selected(): boolean {
+    return this._selected;
+  }
+  set selected(value: boolean) {
+    this._selected = coerceBooleanProperty(value);
+  }
+
+  /** The id of the option, set to a uniqueid if the user does not provide one */
+  @Input()
+  get id(): string {
+    return this._id;
+  }
+  set id(value: string) {
+    this._id = value || this._uniqueid;
+  }
+
+  constructor(@Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+    this._uniqueid = `cdk-option-${nextId++}`;
+    this.id = this.id;
+  }
+
+  /** Toggles the selected state, emits a change event through the injected listbox */
+  toggle() {
+    this.selected = !this.selected;
+    this.listbox._emitChangeEvent(this);
+  }
 
+  static ngAcceptInputType_selected: BooleanInput;
 }
 
+let nextId = 0;
```

Comment:
I would typically put this above the class where it's used

Reply:
Ah true I used to use it in Listbox and neglected moving it.